File: WiSe_23:24_Hausaufgaben/2/pflichtaufgabe1/java_solution/algorithmen/Algorithmen.java
package algorithmen;

public class Algorithmen{  
    /* 
     * Überprüft, ob ein Array aufsteigend sortiert ist.
     */
    public static boolean isSorted(int[] arr){
        for (int i = 0 ; i < arr.length ; i++) {
            for (int j = i+1 ; j < arr.length - 1 ; j++) {
                if (arr[i] >= arr[j]) {
                    return false;
                }
            }
        }
        return true;
    }
    
    
    /* 
     * Diese Funktion sortiert ein gegebenes Array und nutzt Seiteneffekte aus.
     */
    public static void sort(int[] arr){
        for (int i = arr.length - 1 ; i >= 0 ; i--) {
            for (int j = 0 ; j <= i - 1 ; j++) {
                if (arr[j] > arr[j+1]) {
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
    
    }
    
    
    /* 
     * Diese Funktion überprüft mittels binärer Suche, ob das Array arr den Wert val enthält.
     * Darf nur sortierte Arrays erhalten.
     * Gibt nur zurück, ob val in arr enthalten ist.
     */
    public static boolean binarySearch(int[] arr, int val){
        assert isSorted(arr);
        int a=0;
        int b=arr.length;
        while(a != b) {
            int mid = (a + b) / 2;
            if (arr[mid] == val) {
                return true;
            }
            else {
                if (val < arr[mid]) {
                    b = mid;
                }
                else if (val > arr[mid]) {
                    a = mid + 1;
                }
                else {
                    continue;
                }
            }
        }
        return false;
    }
}

File: WiSe_23:24_Hausaufgaben/2/pflichtaufgabe1/python_solution/algorithmen.py
def isSorted(arr):
	return arr == sorted(arr)

def sort(arr):
	for i in range(len(arr)-1 , 1 , -1):
		for j in range(0 , i-1):
			if arr[j] > arr[j+1]:
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
	return arr

def binarySearch(arr , n):
	a = 0
	b = len(arr)
	while a != 0:
		m = (a+b)//2
		if arr[m] == n:
			return True
		else:
			if n < arr[m]:
				b = m
			elif n > arr[m]:
				a = m + 1
			else:
				pass
	return False